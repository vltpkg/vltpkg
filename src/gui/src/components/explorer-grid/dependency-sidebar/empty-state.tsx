import { AnimatePresence, motion } from 'framer-motion'
import { AddDependenciesPopover } from '@/components/explorer-grid/dependency-sidebar/add-dependency.tsx'
import {
  Popover,
  PopoverTrigger,
  PopoverContent,
} from '@/components/ui/popover.tsx'
import { Button } from '@/components/ui/button.tsx'
import { Plus } from 'lucide-react'
import {
  useDependencySidebarStore,
  usePopover,
} from '@/components/explorer-grid/dependency-sidebar/context.tsx'

export const DependencyEmptyState = () => {
  const {
    dependencyPopoverOpen,
    setDependencyPopoverOpen,
    toggleAddDepPopover,
  } = usePopover()

  const dependencies = useDependencySidebarStore(
    state => state.dependencies,
  )

  return (
    <AnimatePresence mode="popLayout" initial={false}>
      {dependencies.length === 0 && (
        <motion.div
          layout
          initial={{ opacity: 0, filter: 'blur(2px)' }}
          animate={{ opacity: 1, filter: 'blur(0px)' }}
          exit={{ opacity: 0, filter: 'blur(2px)' }}
          transition={{
            ease: 'easeInOut',
            duration: 0.25,
          }}
          className="flex min-h-96 w-full cursor-default flex-col items-center justify-center rounded-sm border-[1px] border-dashed border-muted py-12">
          <div className="mb-6 flex justify-center">
            <div className="relative">
              <div className="relative space-y-2">
                <div className="h-12 w-64 rotate-1 transform rounded-lg border-[1px] border-dashed border-muted-foreground/20 bg-muted/30" />

                <div className="-mt-10 h-12 w-64 -rotate-1 transform rounded-lg border-[1px] border-dashed border-muted-foreground/30 bg-muted/50" />

                <div className="relative z-10 -mt-10 flex h-12 w-64 items-center justify-between rounded-lg border-[1px] border-dashed border-muted-foreground/40 bg-background px-4">
                  <div className="flex items-center gap-3">
                    <div className="h-6 w-12 rounded border-[1px] border-dashed border-muted-foreground/30 bg-muted/60" />
                    <div className="h-4 w-24 rounded border-[1px] border-dashed border-muted-foreground/30 bg-muted/60" />
                  </div>
                  <div className="h-4 w-12 rounded border-[1px] border-dashed border-muted-foreground/30 bg-muted/60" />
                </div>
              </div>
            </div>
          </div>

          <div className="flex flex-col items-center justify-center gap-1 text-center">
            <h3 className="text-base font-medium tracking-tight text-foreground">
              No dependencies
            </h3>
            <p className="w-4/5 text-sm font-normal tracking-normal text-muted-foreground">
              Your project's dependencies will appear here, install
              one to see it appear
            </p>
            <Popover
              open={dependencyPopoverOpen}
              onOpenChange={setDependencyPopoverOpen}>
              <PopoverTrigger asChild>
                <Button
                  onClick={toggleAddDepPopover}
                  className="mt-2 h-8 w-fit border-[1px] border-muted">
                  Install a dependency
                  <motion.span
                    animate={{
                      rotate: dependencyPopoverOpen ? 45 : 0,
                    }}>
                    <Plus />
                  </motion.span>
                </Button>
              </PopoverTrigger>
              <PopoverContent
                align="center"
                side="top"
                className="right-0 top-0 w-96 p-0">
                <AddDependenciesPopover />
              </PopoverContent>
            </Popover>
          </div>
        </motion.div>
      )}
    </AnimatePresence>
  )
}
