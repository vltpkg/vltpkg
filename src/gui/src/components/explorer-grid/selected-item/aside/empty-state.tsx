import { AnimatePresence, motion } from 'framer-motion'
import { Package, Search } from 'lucide-react'
import { InlineCode } from '@/components/ui/inline-code.tsx'

export const AsideOverviewEmptyState = () => {
  return (
    <AnimatePresence>
      <motion.div
        layout
        initial={{ opacity: 0, filter: 'blur(2px)' }}
        animate={{ opacity: 1, filter: 'blur(0px)' }}
        exit={{ opacity: 0, filter: 'blur(2px)' }}
        transition={{
          ease: 'easeInOut',
          duration: 0.25,
        }}
        className="mt-4 flex min-h-96 w-full cursor-default flex-col items-center justify-center rounded-xl border-[1px] border-dashed border-muted bg-card py-12 md:mt-0">
        <div className="mb-6 flex justify-center">
          <div className="relative">
            <div className="flex size-28 items-center justify-center rounded-full">
              <Package
                className="size-10 text-muted-foreground"
                strokeWidth={1}
              />

              <div className="absolute bottom-1 right-1 flex size-8 items-center justify-center rounded-full bg-black dark:bg-white">
                <Search className="z-1 relative size-4 text-white dark:text-black" />
              </div>

              <motion.div
                animate={{ opacity: [0.9, 1, 0.9] }}
                transition={{
                  repeat: Infinity,
                  duration: 2,
                  ease: 'easeInOut',
                }}
                className="border-neutral-150 absolute z-[-1] size-28 rounded-full border"
              />
              <motion.div
                animate={{ opacity: [0, 1, 0] }}
                transition={{
                  repeat: Infinity,
                  duration: 2,
                  delay: 0.5,
                  ease: 'easeInOut',
                }}
                className="border-neutral-150 absolute z-[-1] size-36 rounded-full border"
              />
              <motion.div
                animate={{ opacity: [0, 1, 0] }}
                transition={{
                  repeat: Infinity,
                  duration: 2,
                  delay: 0.75,
                  ease: 'easeInOut',
                }}
                className="absolute z-[-1] size-44 rounded-full border"
              />
            </div>
          </div>
        </div>
        <div className="flex flex-col items-center justify-center gap-1 text-center">
          <h3 className="text-base font-medium tracking-tight text-foreground">
            No Overview Available
          </h3>
          <p className="w-4/5 text-pretty text-sm font-normal tracking-normal text-muted-foreground">
            This package does not have any additional metadata. Try
            adding some fields into{' '}
            <InlineCode>package.json</InlineCode>
          </p>
        </div>
      </motion.div>
    </AnimatePresence>
  )
}
