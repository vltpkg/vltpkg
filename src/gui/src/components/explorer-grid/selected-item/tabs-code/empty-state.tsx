import { FileCode2 } from 'lucide-react'

export const EmptyState = () => {
  return (
    <div className="flex h-64 w-full items-center justify-center px-6 py-4">
      <div className="flex flex-col items-center justify-center gap-3 text-center">
        <div className="relative flex size-32 items-center justify-center rounded-full bg-secondary/60">
          <FileCode2
            className="absolute z-[1] size-14 text-neutral-500"
            strokeWidth={1}
          />
        </div>
        <p className="w-2/3 text-pretty text-sm text-muted-foreground">
          The code for this project could not be loaded.
        </p>
      </div>
    </div>
  )
}
