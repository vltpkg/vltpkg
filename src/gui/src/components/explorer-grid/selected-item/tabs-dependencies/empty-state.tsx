import type { LucideIcon } from 'lucide-react'

export const EmptyState = ({
  icon: Icon,
  message,
}: {
  icon: LucideIcon
  message: string
}) => {
  return (
    <div className="flex h-64 w-full cursor-default items-center justify-center px-6 py-4">
      <div className="flex flex-col items-center justify-center gap-3 text-center">
        <div className="relative flex size-32 items-center justify-center rounded-full bg-secondary/60">
          <Icon
            className="absolute z-[3] size-14 text-neutral-500"
            strokeWidth={1.25}
          />
        </div>
        <p className="w-2/3 text-pretty text-sm text-muted-foreground">
          {message}
        </p>
      </div>
    </div>
  )
}
