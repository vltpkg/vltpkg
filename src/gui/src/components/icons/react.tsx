export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
