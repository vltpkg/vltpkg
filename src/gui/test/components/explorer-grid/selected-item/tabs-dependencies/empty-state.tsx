import { test, expect, vi, afterEach } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { Smile } from 'lucide-react'
import { EmptyState } from '@/components/explorer-grid/selected-item/tabs-dependencies/empty-state.tsx'

vi.mock('lucide-react', () => ({
  Smile: 'gui-smile-icon',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  cleanup()
})

test('EmptyState renders with basic options', () => {
  const Container = () => {
    return <EmptyState icon={Smile} message="empty state test" />
  }

  const { container } = render(<Container />)
  expect(container.innerHTML).toMatchSnapshot()
})
