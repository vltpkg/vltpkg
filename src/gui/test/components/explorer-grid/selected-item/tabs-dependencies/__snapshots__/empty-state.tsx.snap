// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyState renders with basic options 1`] = `

<div class="flex h-64 w-full cursor-default items-center justify-center px-6 py-4">
  <div class="flex flex-col items-center justify-center gap-3 text-center">
    <div class="relative flex size-32 items-center justify-center rounded-full bg-secondary/60">
      <gui-smile-icon
        classname="absolute z-[3] size-14 text-neutral-500"
        strokewidth="1.25"
      >
      </gui-smile-icon>
    </div>
    <p class="w-2/3 text-pretty text-sm text-muted-foreground">
      empty state test
    </p>
  </div>
</div>

`;
